### PR TITLE
Add types for outliers

### PR DESCRIPTION
**Pull request by glennreyes**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/outliers/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for outliers 0.0
+// Project: https://github.com/MatthewMueller/outliers
+// Definitions by: Glenn Reyes <https://github.com/glennreyes>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function outliers(input: number[]): number[];
+declare function outliers(path?: string): () => unknown[];
```
Comment (amcasey) [inline types/outliers/index.d.ts:7]: It looks like this actually returns an `Array.prototype.filter` predicate: `(element: number, index: number, array: number[]) => boolean`.

Comment (amcasey) [reply types/outliers/index.d.ts:7]: https://github.com/matthewmueller/outliers/blob/9d9725ce75b55018a0b25f93d92538d7ff24b36c/index.js#L26

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/outliers/index.d.ts (hunk 1 above)
Comment (glennreyes) [reply types/outliers/index.d.ts:7]: Thanks, will update! 👍

**Review by amcasey — APPROVED** [review]

**Review by glennreyes — COMMENTED** [review]
Ready to merge

**Comment by glennreyes** [comment]
Ready to merge

**Pull request merged**